### PR TITLE
Selenium: cover the Workspaces page by selenium test 

### PR DESCRIPTION
**Pull request by SkorikSergey**
### What does this PR do?
This PR cover the Workspaces page by selenium test.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/8131

### CODE REVIEW

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/Workspaces.java [hunk 1]
```diff
@@ -15,59 +15,202 @@
...
-    String WORKSPACES_LIST = "//ng-transclude[@class='che-list-content']";
-    String WORKSPACE_ITEM_NAME =
-        "//div[contains(@class, 'workspace-name-clip')]//div[contains(@data-str, '%s')]";
-    String TOOLBAR_TITLE_NAME =
-        "//div[contains(@class,'che-toolbar')]//span[contains(text(),'%s')]";
-    String ADD_WORKSPACE_BTN =
-        "//a[contains(@ng-href, 'create-workspace')]/span[text()='Add Workspace']";
+    String TOOLBAR = "Workspaces";
+    String DOCUMENTATION_LINK = "//a[@ng-href='/docs/devops/intro/index.html']";
+    String ADD_WORKSPACE_BTN = "//che-button-primary[@che-button-title='Add Workspace']";
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/Workspaces.java]: Does button have name of id attribute? If doesn't, it's better to add it.

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/Workspaces.java [hunk 2]
```diff
@@ -76,6 +219,13 @@ public void selectWorkspaceItemName(String wsName) {
         .click();
   }
 
+  public void waitWorkspaceIsPresent(String nameWorkspace) {
+    new WebDriverWait(seleniumWebDriver, ELEMENT_TIMEOUT_SEC)
+        .until(
+            visibilityOfElementLocated(
+                By.xpath(format(Locators.WORKSPACE_ITEM_NAME, nameWorkspace))));
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/Workspaces.java]: nameWorkspace > workspaceName

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/WorkspacesListTest.java [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    Assert.assertFalse(workspaces.isWorkspaceChecked(workspaceName1));
+  }
+
+  @Test
+  public void checkWorkspaceActions() {
+    // open the Config page of the test workspace
+    workspaces.clickOnWorkspaceConfigureButton(workspaceName1);
+    workspaceDetails.waitToolbarTitleName(workspaceName1);
+    workspaceConfig.waitConfigForm();
+    seleniumWebDriver.navigate().back();
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/WorkspacesListTest.java]: going back looks strange here. What does this step test? If it helps to check on workspace name appearance, it's better to go to workspace list by direct link in Dashboard.

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/WorkspacesListTest.java [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    workspaces.waitWorkspaceIsPresent(workspaceName1);
+
+    // open the Projects page of the test workspace
+    dashboard.selectWorkspacesItemOnDashboard();
+    workspaces.clickOnWorkspaceAddProjectButton(workspaceName2);
+    workspaceDetails.waitToolbarTitleName(workspaceName2);
+    workspaceProjects.waitProjectIsPresent(WEB_JAVA_SPRING);
+    seleniumWebDriver.navigate().back();
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/WorkspacesListTest.java]: please, see https://github.com/eclipse/che/pull/8392/files#r162891028

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/WorkspacesListTest.java [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+    workspaces.waitWorkspaceIsPresent(workspaceName1);
+
+    // check statuses of the created workspaces
+    workspaces.waitWorkspaceStatusIs(workspaceName1, Statuses.STOPPED);
+    workspaces.waitWorkspaceStatusIs(workspaceName2, Statuses.STOPPED);
+    workspaces.waitWorkspaceStatusIs(workspaceName3, Statuses.STOPPED);
+
+    // stop the workspace by the Actions button and check its status is RUNNING
+    dashboard.selectWorkspacesItemOnDashboard();
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/WorkspacesListTest.java]: Why do we need to select ALL workspace items? Or which workspace item are we selecting here?

**Comment by dmytro-ndp** [comment]
Class name **CreateWorkspace**  > **NewWorkspace** - due to the name of page in Dashboard

File: dashboard/src/app/workspaces/list-workspaces/workspace-item/workspace-item.html [hunk 6]
```diff
@@ -53,15 +56,15 @@
            class="che-list-item-name workspace-item-ram"
            ng-click="workspaceItemCtrl.redirectToWorkspaceDetails();">
         <span class="che-xs-header noselect" hide-gt-xs>RAM</span>
-        <span class="workspace-consumed-value">{{workspaceItemCtrl.getMemoryLimit(workspaceItemCtrl.workspace)}}</span>
+        <span class="workspace-consumed-value" name="workspaceRamValue">{{workspaceItemCtrl.getMemoryLimit(workspaceItemCtrl.workspace)}}</span>
```
Comment (dmytro-ndp) [inline dashboard/src/app/workspaces/list-workspaces/workspace-item/workspace-item.html]: Why don't we use "**id**" attribute here instead of "**name**"?

Comment (SkorikSergey) [reply dashboard/src/app/workspaces/list-workspaces/workspace-item/workspace-item.html]: This RAM value exists for every workspace in list.

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/Workspaces.java [hunk 7]
```diff
@@ -15,39 +15,70 @@
...
-    String WORKSPACE_ITEM_NAME =
-        "//div[contains(@class, 'workspace-name-clip')]//div[contains(@data-str, '%s')]";
-    String TOOLBAR_TITLE_NAME =
-        "//div[contains(@class,'che-toolbar')]//span[contains(text(),'%s')]";
-    String ADD_WORKSPACE_BTN =
-        "//a[contains(@ng-href, 'create-workspace')]/span[text()='Add Workspace']";
+    String TOOLBAR = "Workspaces";
+    String DOCUMENTATION_LINK = "//a[@ng-href='/docs/devops/intro/index.html']";
+    String ADD_WORKSPACE_BTN = "//che-button-primary[@che-button-title='Add Workspace']";
+    String DELETE_WORKSPACE_BTN = "//che-button-primary[@che-button-title='Delete']";
```
Comment (akurinnoy) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/Workspaces.java]: @SkorikSergey 
this is the 'Delete' button, you may use the existing class or add any necessary attribute as well
https://github.com/eclipse/che/blob/03e1969339490799dddcd330f3ff4874e2a61b5b/dashboard/src/components/widget/list/che-list-header.html#L31-L35

File: dashboard/src/app/workspaces/list-workspaces/workspace-item/workspace-item.html (hunk 6 above)
Comment (dmytro-ndp) [reply dashboard/src/app/workspaces/list-workspaces/workspace-item/workspace-item.html]: I see, thank you!

Comment (benoitf) [reply dashboard/src/app/workspaces/list-workspaces/workspace-item/workspace-item.html]: is that name shouldn't follow convention of Kebab case vs camelCase like for id where we have "-" ?

**Comment by SkorikSergey** [comment]
ci-build

**Comment by SkorikSergey** [comment]
ci-test

**Review by dmytro-ndp — APPROVED** [review]

**Review by akurinnoy — APPROVED** [review]

**Pull request merged**